src/components/draft/editor/FileSelector.tsx

```diff
@@ -0,0 +1,65 @@
+import { List, ListItemButton, ListItemText } from '@mui/material';
+import useEditorStore, {
+    editorStoreSelectors,
+} from 'components/draft/editor/Store';
+import { FormattedMessage } from 'react-intl';
+import { TABLES } from 'services/supabase';
+import { useQuery, useSelect } from 'supabase-swr';
+
+type Props = {
+    draftId: string;
+};
+
+interface DraftSpecFileList {
+    catalog_name: string;
+    spec_type: string;
+    draft_id: string;
+}
+
+const DRAFT_SPECS_FILE_LIST_QUERY = `
+    catalog_name,
+    spec_type,
+    draft_id
+`;
+
+function EditorFileSelector({ draftId }: Props) {
+    const setCurrentCatalog = useEditorStore(
+        editorStoreSelectors.setCurrentCatalog
+    );
+    const currentCatalog = useEditorStore(editorStoreSelectors.currentCatalog);
+
+    const draftSpecQuery = useQuery<DraftSpecFileList>(
+        TABLES.DRAFT_SPECS,
+        {
+            columns: DRAFT_SPECS_FILE_LIST_QUERY,
+            filter: (query) => query.eq('draft_id', draftId),
+        },
+        []
+    );
```

Comment:
Just let the file selector query it's own list of files as the selector does not really care about the content of files.